app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -118,6 +118,10 @@ if [ -f /etc/redhat-release ]; then
 }
 
     <% if @force -%>
+    if [ -x "$(command -v subscription-manager)" ] ; then
+      subscription-manager clean
```

Comment:
I would call `subscription-manager unregister || true` before that, to let the old Katello know that this client is gone.

`|| true` because it can fail if the old Katello is unreachable or anything.